### PR TITLE
Additions and fixes for inference.py

### PR DESCRIPTION
**Pull request by amorehead**
* Allows one to produce ligand __and__ protein PDB file outputs.
* Allows one to specify the index of the particular CUDA device to use.
* Adds a preliminary means of rank-ordering protein and ligand file outputs, by default, by ligand plDDT (if available) or protein plDDT otherwise.
* @zrqiao, please advise on this implementation of plDDT output ranking. Does it make sense to rank protein-ligand output pairs, by default, by the predicted ligand plDDT (or instead the predicted protein plDDT - which is more informative for ranking do you think?)

### CODE REVIEW

File: README.rst
Comment (zrqiao) [inline README.rst:45]: I would gently push back on this as the device index can be controlled by the CUDA_VISIBLE_DEVICES variable. 
Will unblock this PR, but please update readme.md about this option if the code is merged as-is.

**Review by zrqiao — APPROVED** [review]
Thanks for this contribution!
I would gently recommend making the plddt ranking strategy a CLI option, probably defaulting to using ligand pLDDT to emulate a docking pipeline. 
It needs to be noticed that for the released model checkpoints, the usage case of ranking different model predictions of the same protein-ligand pair (rather than across different complexes) wasn't thoroughly tested, but it would be interesting to see the results.

File: README.rst
Comment (amorehead) [reply README.rst:45]: I can understand the `CUDA_VISIBLE_DEVICES` use case. I personally don't use this approach to selecting GPUs, however, I realize many others do. I'll revert this device index change.

File: neuralplexer/inference.py
Comment (amorehead) [inline neuralplexer/inference.py:226]: @zrqiao, per your comment about making ligand lDDT ranking (e.g., for docking) a CLI option, the `args.rank_outputs_by_confidence` flag now allows users to decide whether or not they would like the predicted complexes for each complex input to be ranked.

File: README.rst
Comment (amorehead) [reply README.rst]: Reverted in [this commit](https://github.com/zrqiao/NeuralPLexer/pull/11/commits/9e07a93c31380ab78021450cdce8c620a422400e).

**Comment by amorehead** [comment]
@zrqiao, if these latest changes look good to you, feel free to merge them. It seems that only those with write access to this repository can merge PRs.

**Comment by zrqiao** [comment]
> @zrqiao, if these latest changes look good to you, feel free to merge them. It seems that only those with write access to this repository can merge PRs.

Thanks @amorehead, LGTM!
I am also adding you to the list of contributors.

**Pull request merged**
